src/main/java/org/jenkinsci/plugins/envinject/EnvInjectJobProperty.java

```diff
@@ -163,45 +181,12 @@ public String getHelpFile() {
 
         @Override
         public EnvInjectJobProperty newInstance(StaplerRequest req, JSONObject formData) throws FormException {
-            Object onObject = formData.get("on");
-
-            if (onObject != null) {
-                EnvInjectJobProperty envInjectJobProperty = new EnvInjectJobProperty();
-                EnvInjectJobPropertyInfo info = req.bindParameters(EnvInjectJobPropertyInfo.class, "envInjectInfoJobProperty.");
-                envInjectJobProperty.setInfo(info);
-                envInjectJobProperty.setOn(true);
-                if (onObject instanceof JSONObject) {
-                    JSONObject onJSONObject = (JSONObject) onObject;
-                    envInjectJobProperty.setKeepJenkinsSystemVariables(onJSONObject.getBoolean("keepJenkinsSystemVariables"));
-                    envInjectJobProperty.setKeepBuildVariables(onJSONObject.getBoolean("keepBuildVariables"));
-                    envInjectJobProperty.setOverrideBuildParameters(onJSONObject.getBoolean("overrideBuildParameters"));
-
-                    //Process contributions
-                    setContributors(req, envInjectJobProperty, onJSONObject);
-
-                    return envInjectJobProperty;
-                }
+            if (formData.optBoolean("on")) {
+                return (EnvInjectJobProperty)super.newInstance(req, formData);
```

Comment:
I copied the code from OptionalJobProperty that isn't available in the current core dep.
I was first trying to remove the on field completely and just check instance, but there are some logic in the backend using it, so I let it be.